src/gggenome.jl

```diff
@@ -0,0 +1,136 @@
+# GGGenome
+# ======
+#
+# APIs for GGGenome.
+#
+# This file is a part of BioJulia.
+# License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+"""
+GGGenome module.
+The `GGGenome` module provides a programming interface to GGGenome, 
+a ultrafast DNA sequence search service, hosted by DBCLS.
+Two functions are exported from this module:
+1. `gggenome`: DNA sequence search
+2. `gggenome_dblist`: List of available databases
+See "GGGenome Help" (https://gggenome.dbcls.jp/en/help.html) for more details.
+"""
+module GGGenome
+
+export
+    gggsearch,
+    gggdbs
+
+
+import HTTP
+
+const baseURL = "https://gggenome.dbcls.jp/"
+const dblistURL = "https://raw.githubusercontent.com/meso-cacase/GGGenome/master/DBlist.pm"
+
+
+# APIs of GGGenome
+# -------------------
+
+"""
+    gggsearch(query; params...)
+Retrieve results of gggenome search of a query sequence.
+
+# Arguments
+## Required 
+- `query::String`: Nucleotide sequence, case insensitive.
+
+## Optional
+- `db::String`: Target database name. hg19 if not specified. Full list of databases: https://gggenome.dbcls.jp/en/help.html#db_list
+- `k::Integer`: Maximum number of mismatches/gaps. 0 if not specified.
+- `strand::String`: '+' ('plus') or '-' ('minus') to search specified strand only.
+- `format::String`: [html|txt|csv|bed|gff|json]. html if not specified.
+- `timeout::Integer`
+- `output::String`: If "toString", a String object is returned. If "extractTopHit", a String object containing only top hit is returned (Currently, only works with format="txt"). Otherwise, A HTTP.Messages.Response object is returned.
+- `show_url::Bool`: If true, print URL of REST API.
+"""
+function gggsearch(query; timeout=5, params...)
+    params = Dict(params)
+    url = generate_url(query, params)
+    if haskey(params, :show_url) && params[:show_url] == true
+        println(url)
+    end
+    res = HTTP.request("GET", url, timeout=timeout)
+
+    if haskey(params, :output)
+        if params[:output] == "toString"
+            return gggenomeToString(res)
+        elseif params[:output] == "extractTopHit" && haskey(params, :format) && params[:format] == "txt"
+            return extractTopHit(gggenomeToString(res))
+        else
+            return res
```

Comment:
I think you should throw an `ArgumentError` exception in this case because returning something without messages may lead to hard-to-find bugs due to a trivial typo (e.g., `output="extractTopHits"`).